channel/pab.go

```diff
@@ -0,0 +1,239 @@
+		acc:                 acc,
+		subscriptionUrlBase: subscriptionUrl,
+		pabRemote: pabRemote{
+			pabUrl: pabUrl,
+		},
+	}, nil
+}
+
+// CallEndpoint calls the given endpoint on the remote wallet and decodes the json response into the given result.
+// `result` must be a pointer.
+func (r *pabRemote) CallEndpoint(endpoint string, body interface{}, result interface{}) error {
+	jsonBody, err := json.Marshal(body)
+	if err != nil {
+		return fmt.Errorf("unable to marshal json body: %w", err)
+	}
+	jsonResponse, err := r.callEndpoint(jsonBody, endpoint)
+	if err != nil {
+		return fmt.Errorf("failed to call endpoint: %w", err)
+	}
+	if result == nil {
+		return nil
+	}
+	if err = json.Unmarshal(jsonResponse, result); err != nil {
+		return fmt.Errorf("failed to unmarshal PAB server response: %w", err)
+	}
+	return nil
+}
+
+// callEndpoint issues a request to the given endpoint with the given body.
+func (r *pabRemote) callEndpoint(jsonBody []byte, endpoint string) ([]byte, error) {
+	request, err := http.NewRequest("POST", r.pabUrl.String()+endpoint, bytes.NewBuffer(jsonBody))
+	if err != nil {
+		return nil, fmt.Errorf("unable to prepare http request: %w", err)
+	}
+	request.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{}
+	response, err := client.Do(request)
+	if err != nil {
+		return nil, fmt.Errorf("unable to send http request: %w", err)
+	}
+	defer response.Body.Close()
+	if response.StatusCode != http.StatusOK {
+		errorBody, err := io.ReadAll(response.Body)
+		if err != nil {
+			return nil, fmt.Errorf("failed to interact with PAB server: %s", response.Status)
+		}
+		return nil, fmt.Errorf(
+			"failed to interact with PAB server: %s with error: %s",
+			response.Status,
+			string(errorBody),
+		)
+	}
+	jsonResponse, err := io.ReadAll(response.Body)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read server response: %w", err)
+	}
+	return jsonResponse, nil
+}
+
+func (p *PAB) SetChannelToken(id channel.ID, token types.ChannelToken) error {
+	emptyToken := types.ChannelToken{}
+	if p.tokenMap[id] != emptyToken {
+		return errors.New("channel token already set in pab")
+	}
+	p.tokenMap[id] = token
+	return nil
+}
+
+func (p *PAB) GetChannelToken(id channel.ID) (types.ChannelToken, error) {
+	token := p.tokenMap[id]
+	emptyToken := types.ChannelToken{}
+	if token == emptyToken {
+		return emptyToken, errors.New("channel token not set in pab")
+	} else {
+		return token, nil
+	}
+}
+
+func (p *PAB) activateContract() error {
+	if p.contractInstanceID != "" {
+		return nil
+	}
+	request := wire.MakePerunActivationBody(p.acc.GetCardanoWalletID())
+	var response wire.ContractInstanceID
+	err := p.pabRemote.CallEndpoint(ActivateEndpoint, request, &response)
+	if err != nil {
+		return fmt.Errorf("failed to activate contract: %w", err)
+	}
+	p.contractInstanceID = response.Decode()
+	return nil
+}
+
+func (p *PAB) createSubscription(id channel.ID, isPerunSub bool) (*AdjudicatorSub, error) {
```

Comment:
Docstring and reference the `newAdjudicatorSub` doc to point the user of this _internal_  function to the correct place, explaining why `isPerunSub` is necessary.